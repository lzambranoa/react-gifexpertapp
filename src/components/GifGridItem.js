import React from 'react';

export const GifGridItem = ( {id, title, url} ) => {

    console.log(id, title, url);
    return (
        <div className='card animate__animated animate__fadeInDown animate__delay-2s'>
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}