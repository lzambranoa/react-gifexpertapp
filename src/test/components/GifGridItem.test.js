import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Pruebas en GifGridItem', () => {

    
    const url = 'https://localhost/algo.jpg';
    const title = 'Un titulo';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrar el <GifGridItem /> correctamente', () => {

        
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe  tner un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );
    });

    test('Debe tener una imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('Debe tener en el div una clase animate-fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate_fadeIn') ).toBe( false );
    });
});
