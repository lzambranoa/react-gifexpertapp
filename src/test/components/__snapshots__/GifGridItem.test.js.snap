// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown animate__delay-2s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
